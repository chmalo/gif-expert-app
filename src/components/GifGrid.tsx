import { useEffect, useState } from "react"
import GifGridItem from "./GifGridItem"

const GifGrid = ({ category }: { category: string }) => {
  const [images, setImages] = useState([])

  useEffect(() => {
    getGifs()
  }, [])

  const getGifs = async () => {
    const url = `https://api.giphy.com/v1/gifs/search?q=Rick+and+Morty&limit=10&api_key=rMr9AzqzYbnJiBtEuJOKTN77v6vJN97S`
    const resp = await fetch(url)
    const { data } = await resp.json()
    const gifs = data.map((img: any) => {
      return {
        id: img.id,
        title: img.title,
        url: img.images?.downsized_medium.url,
      }
    })

    setImages(gifs)
  }

  return (
    <>
      <h3> {category} </h3>
      <div className="card-grid">
        {images.map((img: { id: string; title: string; url: string }) => (
          <GifGridItem key={img.id} {...img} />
        ))}
      </div>
    </>
  )
}

export default GifGrid
