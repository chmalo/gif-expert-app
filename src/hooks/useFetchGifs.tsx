import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs"

export const useFetchGifs = (category: string) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  })

  useEffect(() => {
    getGifs(category).then((imgs) =>
      setState({
        data: imgs,
        loading: false,
      })
    )
  }, [category])

  return state
}
