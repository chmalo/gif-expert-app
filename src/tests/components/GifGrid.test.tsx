import { shallow } from "enzyme"
import { describe, expect, test, vi } from "vitest"
import GifGrid from "../../components/GifGrid"
import { useFetchGifs } from "../../hooks/useFetchGifs"
vi.mock("../../hooks/useFetchGifs")

describe("Testing GifGrid", () => {
  const category = "Goku"

  test("it should show correctly", () => {
    // @ts-ignore
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    })

    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot()
  })

  test("it should show items when images are loaded", () => {
    const gifs = [
      {
        id: "ABC",
        url: "http://localhost/cualquier_cosa.jpg",
        title: "Cualquier cosa",
      },
    ]
    // @ts-ignore
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    })
    const wrapper = shallow(<GifGrid category={category} />)

    // expect(wrapper).toMatchSnapshot()
    expect(wrapper.find("p").exists()).toBe(false)
    expect(wrapper.find("GifGridItem").length).toBe(gifs.length)
  })
})
