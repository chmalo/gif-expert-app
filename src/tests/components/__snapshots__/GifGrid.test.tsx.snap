// Vitest Snapshot v1

exports[`Testing GifGrid > it should show correctly 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid animate__animated animate__fadeIn"
  />
</Fragment>
`;

exports[`Testing GifGrid > it should show items when images are loaded 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    Goku
     
  </h3>
  <div
    className="card-grid animate__animated animate__fadeIn"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http://localhost/cualquier_cosa.jpg"
    />
  </div>
</Fragment>
`;
